primitives/runtime/src/generic/digest.rs

```diff
@@ -374,6 +392,42 @@ impl<'a, Hash> DigestItemRef<'a, Hash> {
 	pub fn try_to<T: Decode>(&self, id: OpaqueDigestItemId) -> Option<T> {
 		self.try_as_raw(id).and_then(|mut x| Decode::decode(&mut x).ok())
 	}
+
+	/// Try to match this to a `Self::Seal`, check `id` matches and decode it.
+	///
+	/// Returns `None` if this isn't a seal item, the `id` isn't of variant seal,
+	/// the `id` identifier doesn't match or when the decoding fails.
+	pub fn seal_try_to<T: Decode>(&self, id: OpaqueDigestItemId) -> Option<T> {
+		match (id, self) {
+			(OpaqueDigestItemId::Seal(w), &Self::Seal(v, s)) if v == w =>
+				Decode::decode(&mut &s[..]).ok(),
+			_ => None,
+		}
+	}
+
+	/// Try to match this to a `Self::Consensus`, check `id` matches and decode it.
+	///
+	/// Returns `None` if this isn't a consensus item, the `id` isn't of variant consensus,
+	/// the `id` identifier doesn't match or when the decoding fails.
+	pub fn consensus_try_to<T: Decode>(&self, id: OpaqueDigestItemId) -> Option<T> {
```

Comment:
I think we can use these functions for the BABE digests as well and remove some duplication, can you update that? 👼